src/trace_writer.ml

```diff
@@ -875,10 +883,34 @@ let maybe_stop_filtered_region t ~should_write =
     t.in_filtered_region <- false)
 ;;
 
+let write_event_and_callstack events_writer event callstack =
+  let compression_event =
+    Callstack_compression.compress_callstack
+      Tracing_tool_output.(events_writer.callstack_compression_state)
```

Comment:
If you annotate the argument `events_writer` with a type, you can replace this line with `events_writer.callstack_compression_state`, which I find a little cleaner. You can also remove the type annotation when matching on it.